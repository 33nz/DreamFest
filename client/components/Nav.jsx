import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import { logOff } from '../auth'
import { clearUser } from '../actions/user'
import { IfAuthenticated, IfNotAuthenticated } from './Authenticated'

function Nav (props) {
  const currentPage = props.location.pathname
  let navLinks = null

  switch (currentPage) {
    case '/signin':
      navLinks = (
        <>
          <Link to="/register">Register</Link>
          <Link to="/">Home</Link>
        </>
      )
      break
    case '/register':
      navLinks = (
        <>
          <Link to="/signin">Sign in</Link>
          <Link to="/">Home</Link>
        </>
      )
      break
    default:
      navLinks = (
        <>
          <Link to="/signin">Sign in</Link>
          <Link to="/register">Register</Link>
        </>
      )
  }

  const logOut = () => {
    logOff()
    props.dispatch(clearUser())
  }

  return (
    <nav className="navbar">
      <IfAuthenticated>
        <Link to="/" onClick={logOut}>
            Log out
        </Link>
        <Link to="/">Home</Link>
      </IfAuthenticated>
      <IfNotAuthenticated>{navLinks}</IfNotAuthenticated>
    </nav>
  )
}

export default connect()(Nav)
