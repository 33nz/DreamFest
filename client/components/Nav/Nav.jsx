import React from 'react'
import { Link } from 'react-router-dom'

import { logOff } from '../../auth-utils'

import { IfAuthenticated, IfNotAuthenticated } from '../Auth/Authenticated'

// hardcoded for the time being
const inviteCode = '111A11'

function Nav () {
  return (
    <ul className='main-nav'>
      <IfAuthenticated>
        <li className='main-nav__item'><Link to='/songs'>My tracks</Link></li>
        <li className='main-nav__item'><Link to={'/friends'}>My friends</Link></li>
        <li className='main-nav__item'><Link to={'/profile'}>Edit my profile</Link></li>
        <li className='main-nav__item'>My code: {inviteCode}</li>
        <li className='main-nav__item'><Link to='/' onClick={logOff}>Log Out</Link></li>
      </IfAuthenticated>
      <IfNotAuthenticated>
        <li className='main-nav__item'><Link to='/signIn'>Log in</Link></li>
        <li className='main-nav__item'><Link to='/register'>Register</Link></li>
      </IfNotAuthenticated>
    </ul>

  )
}

export default Nav
