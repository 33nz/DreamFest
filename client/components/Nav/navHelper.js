import { dispatch } from '../../store'
// import { logOff } from '../../auth'
import { clearUser } from '../../actions/user'

export function logOut () {
  // logOff()
  dispatch(clearUser())
}

export function getLinks (page) {
  switch (page) {
    case '/signin':
      return [
        { to: '/register', name: 'Register' },
        { to: '/', name: 'Home' }
      ]

    case '/register':
      return [
        { to: '/signin', name: 'Sign in' },
        { to: '/', name: 'Home' }
      ]

    default:
      return [
        { to: '/signin', name: 'Sign in' },
        { to: '/register', name: 'Register' }
      ]
  }
}
