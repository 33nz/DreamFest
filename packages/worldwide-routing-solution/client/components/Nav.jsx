import React from 'react'
import { Link } from 'react-router-dom'

import continentData from '../../data/continents'

function Nav (props) {
  const continents = Object.keys(continentData)
  const url = props.location.pathname

  return (
    <div>
      <h2>Nav</h2>
      <ul>
        <li><Link to='/'>Home</Link></li>
      </ul>
      <ul>
        {continents.map((name, i) => {
          return <li key={i} className={url.includes(name) ? 'selected' : ''}>
            <Link to={`/continent/${name}`}>{name}</Link>
          </li>
        })}
      </ul>
    </div>
  )
}

export default Nav
