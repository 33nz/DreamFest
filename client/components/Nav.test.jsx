import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { HashRouter as Router } from 'react-router-dom'

import Nav from './Nav'

describe('<Nav> renders correctly', () => {
  it('renders correctly on the "/" route', () => {
    const tree = renderer
      .create(
        <Router>
          <Nav location={{ pathname: '/' }}/>
        </Router>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly on the "/signin" route', () => {
    const tree = renderer
      .create(
        <Router>
          <Nav location={{ pathname: '/signin' }}/>
        </Router>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly on the "/register" route', () => {
    const tree = renderer
      .create(
        <Router>
          <Nav location={{ pathname: '/register' }}/>
        </Router>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

test('<Nav> renders a "register" and a "sign in" link on the "/" route', () => {
  render(
    <Router>
      <Nav location={{ pathname: '/' }}/>
    </Router>)
  expect(screen.getByText('Register')).toHaveTextContent('Register')
  expect(screen.getByText('Sign in')).toHaveTextContent('Sign in')
})

test('<Nav> renders a "register" and a "home" link on the "/signin" route', () => {
  render(
    <Router>
      <Nav location={{ pathname: '/signin' }}/>
    </Router>)
  expect(screen.getByText('Register')).toHaveTextContent('Register')
  expect(screen.getByText('Home')).toHaveTextContent('Home')
})

test('<Nav> renders a "sign in" and a "home" link on the "/register" route', () => {
  render(
    <Router>
      <Nav location={{ pathname: '/register' }}/>
    </Router>)
  expect(screen.getByText('Sign in')).toHaveTextContent('Sign in')
  expect(screen.getByText('Home')).toHaveTextContent('Home')
})
