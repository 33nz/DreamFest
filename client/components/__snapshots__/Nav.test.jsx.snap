// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav> renders correctly renders correctly on the "/" route 1`] = `
<div
  className="nav"
>
  <a
    href="#/signin"
    onClick={[Function]}
  >
    Sign in
  </a>
  <a
    href="#/register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;

exports[`<Nav> renders correctly renders correctly on the "/register" route 1`] = `
<div
  className="nav"
>
  <a
    href="#/signin"
    onClick={[Function]}
  >
    Sign in
  </a>
  <a
    href="#/"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;

exports[`<Nav> renders correctly renders correctly on the "/signin" route 1`] = `
<div
  className="nav"
>
  <a
    href="#/register"
    onClick={[Function]}
  >
    Register
  </a>
  <a
    href="#/"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;
